kratos/includes/model_part.h

```diff
@@ -796,6 +796,23 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
      */
     void AddProperties(PropertiesType::Pointer pNewProperties, IndexType ThisIndex = 0);
 
+    /** Returns the Properties corresponding to it's identifier is defined */
+    bool HasProperties(IndexType PropertiesId, IndexType ThisIndex = 0)
+    {
+        auto pprop_it = GetMesh(ThisIndex).Properties().find(PropertiesId);
+        if(pprop_it != GetMesh(ThisIndex).Properties().end()) { //property does exist
+            return true;
+        } else {
+            if(IsSubModelPart()) {
+                return mpParentModelPart->HasProperties(PropertiesId, ThisIndex);
```

Comment:
It checks the parent model part

Reply:
really not clear to me of what is the rationale.

if i have a model_part and a submodelpart

and i ask to the submodelpart if it "HasProperties" than it should check on itself it it has it. If it doesn't it should not add it nor ask to the father... 

just imagine the following scenario

mp has property 1
sub doesn't 

you say

      if(sub.HasProperties(1))
         
with your implementation it will say "true", but when you try to Get it you would get a new one, not consistent with the one in the father

Actually in my opinion this method should be **const** since it should not produce any modification onto the properties class

Reply:
OK then

Reply:
Can you please add a test for exactly this?

This will make it very easy to understand for others and also prevent changes in behavior if sth is refactored